[MacOS] WKWebView should have SPI to report whether a PDF is being displayed

aprotyas opened the pull request:
#### dc1ebf3e81905b72da8d9f266b88db88d7847f08
<pre>
[MacOS] WKWebView should have SPI to report whether a PDF is being displayed
<a href="https://bugs.webkit.org/show_bug.cgi?id=273639">https://bugs.webkit.org/show_bug.cgi?id=273639</a>
<a href="https://rdar.apple.com/127442624">rdar://127442624</a>

Reviewed by Tim Horton.

This patch exposes the `_isDisplayingPDF` method on WKWebViewMac, which
clients can query to answer whether the web view is presenting a main
frame PDF.

This also establishes parity with WKWebViewIOS, where the namesake was
added in 149116@main.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _isDisplayingPDF]):

Move getter implementation to this shared location.

* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:

Move property declaration to the shared WKPrivate category.

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _isDisplayingPDF]): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPDFView.mm:
(TEST(WKWebView, IsDisplayingPDF)):

Add an API test exercising the `_isDisplayingPDF` SPI.

Canonical link: <a href="https://commits.webkit.org/278324@main">https://commits.webkit.org/278324@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/09f81a5b981f7556a03ecafb95ca3bc89c6f0560

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50182 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29473 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/2476 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/53440 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/872 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/52485 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/35700 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/464 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40949 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52281 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27150 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43188 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22047 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/24573 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/439 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/8560 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/46560 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/500 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55023 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25278 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/48347 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/26539 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43373 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47362 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/27402 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7254 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/26271 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
```diff
@@ -771,6 +771,8 @@ typedef NS_OPTIONS(NSUInteger, WKDisplayCaptureSurfaces) {
 // Defaults to YES; if set to NO, WebKit will draw the grey wash and highlights itself.
 @property (nonatomic, setter=_setUsePlatformFindUI:) BOOL _usePlatformFindUI WK_API_AVAILABLE(macos(10.15));
 
+@property (nonatomic, readonly, getter=_isDisplayingPDF) BOOL _displayingPDF WK_API_AVAILABLE(macos(WK_MAC_TBA));
```
Shouldn't this be the iOS version but moved to a shared place???

aprotyas replied:
Well, yes, but the two impls are distinct, so now we have the property `_displayingPDF` in both the `WKPrivateMac` and the `WKPrivateIOS` categories, with the former implemented in `WKWebViewMac` and the latter implemented in `WKWebViewIOS`.

aprotyas replied:
In hindsight, this fragmented declaration is markedly worse than moving to a shared place, so I'll just do that instead.

aprotyas replied:
Fixed in https://github.com/WebKit/WebKit/commit/030e8c9ccab5aff0053a0ba85d32d9ffdeb58b9a

aprotyas committed:
[MacOS] WKWebView should have SPI to report whether a PDF is being displayed
https://bugs.webkit.org/show_bug.cgi?id=273639
rdar://127442624

Reviewed by Tim Horton.

This patch exposes the `_isDisplayingPDF` method on WKWebViewMac, which
clients can query to answer whether the web view is presenting a main
frame PDF.

This also establishes parity with WKWebViewIOS, where the namesake was
added in 149116@main.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _isDisplayingPDF]):

Move getter implementation to this shared location.

* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:

Move property declaration to the shared WKPrivate category.

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _isDisplayingPDF]): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPDFView.mm:
(TEST(WKWebView, IsDisplayingPDF)):

Add an API test exercising the `_isDisplayingPDF` SPI.

Canonical link: https://commits.webkit.org/278324@main
